scipy/signal/_signaltools.py

```diff
@@ -4625,3 +4625,142 @@ def decimate(x, q, n=None, ftype='iir', axis=-1, zero_phase=True):
         sl[axis] = slice(None, None, q)
 
     return y[tuple(sl)]
+
+def envelope(x,N = None,method = 'analytic'):
```

Comment:
Please follow PEP8 guidelines 

```suggestion
def envelope(x, N=None, method = 'analytic'):
```